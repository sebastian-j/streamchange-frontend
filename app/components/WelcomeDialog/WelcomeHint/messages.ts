import { defineMessages } from 'react-intl';

export const scope = 'streamchange.components.WelcomeDialog';

export default defineMessages({
  buttonLabel: {
    id: `${scope}.WelcomeHint.button.label`,
    defaultMessage: 'Draw another curiosity'
  },
  hintTitle: {
    id: `${scope}.WelcomeHint.title`,
    defaultMessage: 'Did you know...',
  },
  holiday1402: {
    id: `${scope}.holidays.14-02`,
    defaultMessage: 'February 14 - Valetine\'s Day',
  },
  holiday1403: {
    id: `${scope}.holidays.14-03`,
    defaultMessage: 'March 14 - Pi Day',
  },
  holiday2004: {
    id: `${scope}.holiday.20-04`,
    defaultMessage: 'April 20 - Cannabis Day',
  },
  holiday1705: {
    id: `${scope}.holiday.17-05`,
    defaultMessage:
      'May 17 - International Day Against Homophobia, Biphobia and Transphobia',
  },
  holidayJune: {
    id: `${scope}.holiday.x-06`,
    defaultMessage: 'June - Pride Month',
  },
  holiday2608: {
    id: `${scope}.holiday.26-08`,
    defaultMessage: 'August 26 - International Dog Day',
  },
  holiday2309: {
    id: `${scope}.holiday.23-09`,
    defaultMessage: 'First Day of Fall',
  },
  holiday0110: {
    id: `${scope}.holiday.01-10`,
    defaultMessage: 'October 1 - World Vegetarian Day',
  },
  holiday1110: {
    id: `${scope}.holiday.11-10`,
    defaultMessage: 'October 11 - Coming Out Day',
  },
  holiday3110: {
    id: `${scope}.holiday.31-10`,
    defaultMessage: 'October 31 - Halloween',
  },
  holiday0612: {
    id: `${scope}.holiday.06-12`,
    defaultMessage: 'December 6 - Santa Claus',
  },
  holiday2512: {
    id: `${scope}.holiday.25-12`,
    defaultMessage: 'December 25, 26 - Christmas',
  },
  holiday3112: {
    id: `${scope}.holiday.31-12`,
    defaultMessage: 'December 31 - New Year\'s Eve',
  },
});
