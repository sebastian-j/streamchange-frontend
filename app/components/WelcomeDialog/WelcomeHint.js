import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import LightBulbImg from './light-bulb.png';
import messages from './messages';
import { HINTS } from '../../config';

const Wrapper = styled.div`
  background-color: white;
  display: flex;
  flex-direction: row;
  box-shadow: 0 11px 15px -7px rgba(0, 0, 0, 0.2),
    0 24px 38px 3px rgba(0, 0, 0, 0.14), 0 9px 46px 8px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  margin-top: 20px;
  max-width: 600px;
`;

const DialogTitle = styled.span`
  font-size: 1.2rem;
  font-weight: bold;
  display: block;
  margin: 10px;
`;

const DialogContent = styled.span`
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  display: block;
  margin: 10px;
`;

const Button = styled.button`
  background: none;
  border: none;
  outline: none;
`;

const Image = styled.img`
  height: 140px;
`;

const WelcomeHint = () => {
  const [hint, setHint] = useState('');

  useEffect(() => {
    setHint(HINTS[Math.floor(Math.random() * HINTS.length)]);
  }, []);

  const changeHint = () => {
    setHint(HINTS[Math.floor(Math.random() * HINTS.length)]);
  };

  return (
    <Wrapper>
      <Button onClick={changeHint}>
        <Image src={LightBulbImg} alt="light bulb" />
      </Button>
      <div>
        <DialogTitle>
          <FormattedMessage {...messages.hintTitle} />
        </DialogTitle>
        <DialogContent>{hint}</DialogContent>
      </div>
    </Wrapper>
  );
};

export default WelcomeHint;
