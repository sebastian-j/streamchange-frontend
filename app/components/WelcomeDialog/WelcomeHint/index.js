import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import CannabisImg from '../assets/cannabis.webp';
import ChristmasImg from '../assets/christmas.webp';
import DogImg from '../assets/dog.webp';
import FallImg from '../assets/fall.webp';
import FireworksImg from '../assets/fireworks.webp';
import HalloweenImg from '../assets/halloween.webp';
import LightBulbImg from '../assets/light-bulb.webp';
import PiImg from '../assets/pi.webp';
import RainbowFlagImg from '../assets/rainbow-flag.webp';
import SantaClausImg from '../assets/santa claus.webp';
import ValentineImg from '../assets/valentine.webp';
import VeganImg from '../assets/vegan.webp';
import { DialogContent } from './components/DialogContent';
import { DialogTitle } from './components/DialogTitle';
import { HintPanel } from './components/HintPanel';
import { HolidayPanel } from './components/HolidayPanel';
import { ImageButton } from './components/ImageButton';
import { Wrapper } from './components/Wrapper';
import messages from './messages';
import { HINTS } from '../../../config';

const WelcomeHint = () => {
  const [hint, setHint] = useState('');
  const [image, setImage] = useState(LightBulbImg);
  const [holiday, setHoliday] = useState(null);

  const holidaysDates = [
    [/14-02/, ValentineImg, 'holiday1402'],
    [/14-03/, PiImg, 'holiday1403'],
    [/20-04/, CannabisImg, 'holiday2004'],
    [/17-05/, RainbowFlagImg, 'holiday1705'],
    [/\d-06/, RainbowFlagImg, 'holidayJune'],
    [/26-08/, DogImg, 'holiday2608'],
    [/23-09/, FallImg, 'holiday2309'],
    [/01-10/, VeganImg, 'holiday0110'],
    [/11-10/, RainbowFlagImg, 'holiday1110'],
    [/31-10/, HalloweenImg, 'holiday3110'],
    [/06-12/, SantaClausImg, 'holiday0612'],
    [/(?:25|26)-12/, ChristmasImg, 'holiday2512'],
    [/31-12/, FireworksImg, 'holiday3112'],
  ];

  const getToday = () => {
    const d = new Date();
    let day = `${d.getDate()}`;
    let month = `${d.getMonth() + 1}`;

    if (month.length < 2) month = `0${month}`;
    if (day.length < 2) day = `0${day}`;

    return [day, month].join('-');
  };

  useEffect(() => {
    setHint(HINTS[Math.floor(Math.random() * HINTS.length)]);
    for (let i = 0; i < holidaysDates.length; i += 1) {
      if (holidaysDates[i][0].test(getToday())) {
        setImage(holidaysDates[i][1]);
        setHoliday(holidaysDates[i][2]);
        break;
      }
    }
  }, []);

  const changeHint = () => {
    setHint(HINTS[Math.floor(Math.random() * HINTS.length)]);
  };

  return (
    <Wrapper>
      <HintPanel>
        <ImageButton onClick={changeHint}>
          <img src={image} alt="light bulb" />
        </ImageButton>
        <div>
          <DialogTitle>
            <FormattedMessage {...messages.hintTitle} />
          </DialogTitle>
          <DialogContent>{hint}</DialogContent>
        </div>
      </HintPanel>
      {holiday && (
        <HolidayPanel>
          <FormattedMessage {...messages[holiday]} />
        </HolidayPanel>
      )}
    </Wrapper>
  );
};

export default WelcomeHint;
