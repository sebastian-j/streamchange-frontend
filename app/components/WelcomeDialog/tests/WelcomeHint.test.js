import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import WelcomeHint from '../WelcomeHint';

const renderer = new ShallowRenderer();

describe('<WelcomeHint />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<WelcomeHint />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
