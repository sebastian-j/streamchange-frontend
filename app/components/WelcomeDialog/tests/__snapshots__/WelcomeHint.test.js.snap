// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeHint /> should render and match the snapshot 1`] = `
<ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    <img
      alt="light bulb"
      height="140px"
      src="/static/light-bulb.png"
    />
  </ForwardRef>
  <div>
    <ForwardRef>
      <FormattedMessage
        defaultMessage="Did you know..."
        id="streamchange.components.WelcomeDialog.WelcomeHint.title"
        values={Object {}}
      />
    </ForwardRef>
    <ForwardRef>
      
    </ForwardRef>
  </div>
</ForwardRef>
`;
